### PR TITLE
Add REST API resources and missing constants

### PR DESCRIPTION
**Pull request by saddieeiddas**
- Add `gender` constants
- Add `attributeType` constants
- Add `faction` constants
- Add `Characters` API resource
  https://api.camelotunchained.com/#api-Characters
- Add `GameData` API Resource
  https://api.camelotunchained.com/#api-Game_Data
- Add `Servers` API Resource
  https://api.camelotunchained.com/#api-Servers

Could be used as follows:

```js
import {restAPI} from 'camelot-unchained';

restAPI.getServers().then((servers) => {
  console.log(servers);
});

restAPI.getCharacters().then((characters) => {
  characters.forEach((character) => {
    // restAPI.Characters.deleteCharacterOnShard(character.shardID, character.id)
  });
});

restAPI.createCharacter(1, 4, {
  "name":"MrAwesome",
  "faction": 2,
  "race": 5,
  "gender": 1,
  "archetype": 3,
  "attributes":{
    "Strength":0,
    "Dexterity":0,
    "Agility":20,
    "Vitality":0,
    "Endurance":0,
    "Attunement":0,
    "Will":0,
    "Faith":0,
    "Resonance":0,
    "Eyesight":0
  },
  "boons":{
    "5429de0eda9beb2c3c3dd32b":1
  },
  "banes":{
    "5429de13da9beb2c3c3dd450":1,
    "5429de13da9beb2c3c3dd451":1,
    "5429de13da9beb2c3c3dd452":2
  }
}).then(function(id) {
  console.log('created character with id ' + id);
}).catch(function(error) {
});
```

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/Camelot-Unchained-Client-Library/pull/17" target="_blank">ReviewNinja

File: src/restapi/resources/Characters.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  return RestClient.postJSON(`/characters/${shardID}/${channelId}`, true, data)
+    .then(RestUtil.getPath)
+    // API is returing the full URL get getCharacterOnShard with the ID, so we will call that path to obtain the newly created character
+    .then((path) => {
+      return RestClient.getJSON(path, true);
+    });
+}
+
+// Response when calling getCharacters or getCharactersOnShard
+export interface CharacterListResponse {
```
Comment (codecorsair) [inline src/restapi/resources/Characters.ts]: This is called SimpleCharacter

File: src/restapi/resources/Characters.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  return RestClient.deleteJSON(`/characters/${shardID}/${characterID}`, true);
+}
+
+// Create a Character
+export function createCharacter(shardID: number, channelId: channelId, data: CharacterCreateRequest) {
+  return RestClient.postJSON(`/characters/${shardID}/${channelId}`, true, data)
+    .then(RestUtil.getPath)
+    // API is returing the full URL get getCharacterOnShard with the ID, so we will call that path to obtain the newly created character
+    .then((path) => {
+      return RestClient.getJSON(path, true);
```
Comment (codecorsair) [inline src/restapi/resources/Characters.ts]: I don't like having a second hidden call in here. In our current use case with the UI we do not even need to make this second call as we would not even use the full character data for anything.  The UI which made the call to create character it closed as soon as the character is created successfully and the user is logged into the server.

File: src/restapi/RestAPI.ts [hunk 3]
```diff
@@ -17,16 +19,26 @@ export function getCraftedAbilities(loginToken: string, characterID: string) {
...
   return RestClientLegacy.getJSON('game/players');
 }
 
 // TODO update this to use new Rest Client
-export function postPlotPermissions(query: Object) {
+function postPlotPermissions(query: Object) {
   return RestClientLegacy.postJSON('plot/modifypermissions', true, false, query);
 }
+
+export {
```
Comment (codecorsair) [inline src/restapi/RestAPI.ts]: I actually prefer exporting each function individually here.

Comment (saddieeiddas) [inline src/restapi/RestAPI.ts]: Do you want all functions to be called directly from `restAPI` without the namespace (`GameData`, `Characters`) etc?

If so, we could re-export the resources:

```
export * from './resources/GameData';
export * from './resources/Servers';
export * from './resources/Characters';
```

Only worry with this is clashes between different resources (but this realistically shouldn't happen).

Would then have to call all the methods without namespace.

`restAPI.Servers.getServers()` would be `restAPI.getServers()`

If this isn't the case, then its more complicated to import then export... we could potentially do:

```
import * as GameDataResource from './resources/GameData';
const GameData = GameDataResource;
export GameData;
```

Not sure if that will work well.

Comment (codecorsair) [inline src/restapi/RestAPI.ts]: I'm fine with directly exporting.

**Comment by codecorsair** [comment]
Looks good. :+1:

**Pull request merged**